packages/reporting-server/README.md

```diff
@@ -88,8 +88,148 @@ postgres://<user>:<password>@<host>/<database>
+
+``` bash
+sudo -i -u postgres
+```
+
+Then you can access the Postgres prompt by typing:
+
+``` bash
+psql
+```
+
+This will log you into the PostgreSQL prompt, and from here you are free to interact with the database management system right away.
+
+We can create new user with the following command
+
+``` bash
+create user <youruser> with encrypted password <yourpassword>;
+
+# E.G: create user myuser with encrypted password 'mypassword';
+```
+
+After the user is created we need to assign the db roles to the user:
+
+``` bash
+grant all privileges on database <yourdatabase> to <youruser>;
+
+# E.G: grant all privileges on database reporting to myuser;
+```
+
+### Set/change user PostgreSQL password
+
+``` bash
+sudo -i -u postgres
+```
+
+Then you can access the Postgres prompt by typing:
+
+``` bash
+psql
+```
+
+To change your password you run the following command:
+
+``` bash
+\password <user>
+# E.g: \password postgres
+```
+
+To exit out of the PostgreSQL prompt, run the following:
+
+``` bash
+\q
+```
+
+
+
+### PostgresSQL URL
+
+Once the above steps are finished you can build your bd_url:
+
+``` 
+postgres://<user>:<password>@<host>/<database>
+```
+
+For more information you can check [this tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-postgresql-on-ubuntu-20-04-quickstart)
+
+
+## Run the application
+
+Now that you have already elected, configured and created a database and set you configurations. You need to run the migrations to create the database tables. We can proceed to apply migrations to create the database tables by running
+
+```bash
+aerich upgrade
+```
+
+After running this command your database should have all the tables defined in the code. Now you can run the application
+
+```bash
+reporting_server
+```
+
+If you want to run the migrations and start the server you could also run:
+
+```bash
+npm run start:dev
+```
+
 # Developers
 
+## Migration
+We are using [aerich](https://github.com/tortoise/aerich) as our database migration tool. That means that the changes to made to the model will not be reflected in the database automatically. You must run the aerich migration command to create a new migration with the changes:
+
+``` bash
+npm run migrate
+```
+
+If there are some changes in the Tortoise models, the previous command will create a migration (sql) file that has your changes. To apply these changes to the database you need to run
+
+``` bash
+npm run apply:migrations 
+```
+
+![aerich](https://user-images.githubusercontent.com/11761240/122826198-d97f2e80-d2b0-11eb-813f-384f4ae61d6b.png)
+
+To change the settings for the url, user and password you can open the [default_config file](https://github.com/open-rmf/rmf-web/blob/main/packages/reporting-server/rest_server/default_config.py) and change the value of the `db_url` with the following format (E.g PostgreSQL):
```

Comment:
Changing the source doesn't look like a good way to change the configs, if you installed from `pip3`, then you need to change system files, and it will be overwritten when you update to a new version.

Reply:
Yeah, but we can override that, as we do it in the example-deployment, right?

Reply:
It's possible to do it but there may be some problems

* Someone that uses `pip3 install reporting-server` will have to find directory where the package is installed (this could be a system dir owned by root), go into the source and change the config.
* The default config will be overwritten when the user upgrades the package, so you have to do the patching again.
* For a docker image, you need to mount a volume on the config file. This might not be obvious so might be good to have some docs explaining how to do it and the location of the config file in the image.
  * For kubernetes, you have to use a configmap and mount it in the correct directory, so it might be good to explain this as well.